src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -109,6 +115,18 @@ private void EndRequest(HttpContext httpContext, long timestamp)
                     telemetry.Success &= successExitCode;
                 }
 
+                IHeaderDictionary requestHeaders = httpContext.Request?.Headers;
```

Comment:
Can you please follow this PR: https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/326 to allow customer to override the `Source`